includes/forms/controllers/class.llms.controller.account.php

```diff
@@ -392,6 +392,7 @@ public function reset_password_link_redirect() {
 			$uid  = $user ? $user->ID : 0;
 			$val  = sprintf( '%1$d:%2$s', $uid, wp_unslash( llms_filter_input_sanitize_string( INPUT_GET, 'key' ) ) );
 
+			( new LLMS_Cache_Helper() )->maybe_no_cache();
```

Comment:
hey @pondermatic I'm struggling to figure out why this is needed here... shouldn't the nocaching get handled automatically due to the `maybe_no_cache()` running automatically against the dashboard page (and it's endpoints)?

Reply:
It's a priority thing, the redirect here happens at `wp|5` while the `maybe_no_cache()` at `wp|10` (iirc).

Reply:
When the reset link URL is requested, `LLMS_Controller_Account::reset_password_link_redirect()` is called from the `wp` action hook with a priority of 1 and then the web browser is redirected before `LLMS_Cache_Helper::maybe_no_cache()` can be called from the `wp` action hook with a priority of 10.

Reply:
Yea, right, but when the page reloads it's on the dashboard and `maybe_no_cache()` should set the cookie at `wp|10`, right?

I feel like I'm demonstrating my stupidity here somehow but on the next page load (the reset page) the `maybe_no_cache()` should load anyway right? Right?

What am I being dumb about here that you both are saying the same thing but not actually addressing my question which is that `maybe_no_cache()` will run when the page reloads (after the redirect) as the next page loads... Right?

Reply:
@gocodebox/engineering bump :arrow_up:

Reply:
Well I hadn't tested it on wpe, I just figured @pondermatic found that before redirecting something was cached by wpe?

Reply:
@thomasplevy,

The "no cache" cookie has to be sent to the web browser before it requests the page that allows setting a new password.

The redirect response to the emailed "Reset Password" link must include a `set-cookie: wordpress_XXX` HTTP header. When the browser follows the redirected location, it sends the cookie which tells WP Engine to not use the server-side cache.

If the web browser's cache is cleared between submitting the user's email address and following the link in the email, [`LLMS_Controller_Account::reset_password_link_redirect()`](https://github.com/gocodebox/lifterlms/blob/ce212de3b02eefa2eef6b740e4950ac83af3a645/includes/forms/controllers/class.llms.controller.account.php#L387) must call `( new LLMS_Cache_Helper() )->maybe_no_cache();` because it redirects and exits before the [`wp`](https://github.com/WordPress/wordpress-develop/blob/18bda14e1e78eba29b2df570583a8b6320d26a58/src/wp-includes/class-wp.php#L771) action hook can call [`LLMS_Cache_Helper::maybe_no_cache()`](https://github.com/gocodebox/lifterlms/blob/5bfa3bc1c02b99db1e97003e51b11d53d978ad8e/includes/class.llms.cache.helper.php#L115).

```mermaid
sequenceDiagram;
participant client AS Web Browser;
participant server AS Web Server;
client ->>+server: GET /dashboard/lost-password/?key=KEY&login=NAME
server ->>client: set-cookie: wordpress_XXX
server->>-client: location: /dashboard/lost-password/?reset-pass=1

client ->>+ server: GET /dashboard/lost-password/?reset-pass=1
client->>server: cookie: wordpress_XXX
server ->>- client: reset password page
```

Reply:
This method and class need changelog notes though